src/back/wgsl/writer.rs

```diff
@@ -79,25 +123,75 @@ impl<W: Write> Writer<W> {
     ///
     /// # Notes
     /// Ends in a newline
-    fn write_function(&mut self, func: &Function) -> BackendResult {
-        write!(self.out, "fn {}(", func.name.as_ref().unwrap())?; // TODO: unnamed function?
+    fn write_function(
+        &mut self,
+        module: &Module,
+        func: &Function,
+        func_ctx: &FunctionCtx<'_>,
+    ) -> BackendResult {
+        // TODO: unnamed function?
+        write!(self.out, "fn {}(", func.name.as_ref().unwrap())?;
+
+        // Write function arguments
+        for (index, arg) in func.arguments.iter().enumerate() {
+            // Write argument attribute if a binding is present
+            if let Some(ref binding) = arg.binding {
+                self.write_attribute(map_binding_to_attribute(binding))?;
+                write!(self.out, " ")?;
+            }
+            // TODO: unnamed args?
+            // Write argument name
+            write!(self.out, "{}: ", arg.name.as_ref().unwrap())?;
```

Comment:
this should use `self.names[key]`, where key depends on whether this is a function or an entry point (can be passed as an argument to `write_function`)